bip-XXXX-Bech32_Encoded_Transaction_Position_References.mediawiki

```diff
@@ -0,0 +1,277 @@
+
+The following strings have a valid Human Readable Part and Bech32 Checksum.
+* <tt>TX1A12UEL5L</tt>
+* <tt>tx1an83characterlonghumanreadablepartthatcontainsthenumber1andtheexcludedcharactersbio1tt5tgs</tt>
+* <tt>tx1abcdef1qpzry9x8gf2tvdw0s3jn54khce6mua7lmqqqxw</tt>
+* <tt>tx11qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqc8247j</tt>
+
+The following list gives invalid TxRef's and the reason for their invalidity.
+* <tt>bc1qw508d6qejxtdg4y5r3zarvary0c5xw7kg3g4ty</tt>: Invalid human-readable part
+* <tt>tx1qw508d6qejxtdg4y5r3zarvary0c5xw7kv8f3t5</tt>: Invalid checksum
+
+=== Allocated Magic Values ===
+
+This appendix contains a table of the allocated or reserved magic codes.
+* If you wish to claim a magic code, please open a pull request adding the code to this list.
+* If this list becomes too popular, we will move it to another location.
+
+==== Reserved Values: ====
+Magic Values 5-bit long:
+{| class="wikitable"
+|'''Magic Value'''
+|'''Starting Value(s)'''
+|'''Display'''
+|'''Description'''
+|-
+|
+|0x0
+|‘q’
+|Reserved
+|-
+|
+|0x1
+|‘p’
+|Reserved
+|-
+|
+|0x2
+|‘z’
+|Reserved
+|-
+|'''Bitcoin'''
+|'''0x3'''
+|'''‘r’'''
+|'''Bitcoin Main Chain'''
+|-
+|'''Litecoin'''
+|'''0x4'''
+|'''‘y’'''
+|'''Litecoin Main Chain'''
+|-
+|
+|0x5
+|‘9’
+|Reserved
+|-
+|'''Testnet'''
+|'''0x6'''
+|'''‘x’'''
+|'''Any Test Network'''
+|-
+|
+|0x7
+|‘8’
+|Reserved
+|}
+Magic Values 10-bit long: (none allocated).
+
+Magic Values 15-bit long: (none allocated).
+
+Magic Values 20-bit long: (none allocated);
+
+=== Standard Display Formats ===
+
+For non-bitcoin related display formats please reference the external SLIP-XXXX "TxRef Non-Bitcoin Display Formats".
+
+=== Bitcoin Display Formats ===
+This appendix contains the specifications for Bitcoin Related Application Display Formats.
+* If you wish to contribute a new display-format, please open a pull request adding to this appendix.
+** For Example. A display format for a Bitcoin Sidechain.
+* We have deliberately not included a Bitcoin display format for the years past 2048.
+** We think it is prudent to design the new display format closer to this date.
+
+==== Bitcoin Display Format 0 ====
+This application display format is defined for all Bitcoin main-network transactions that have:
+* Block Height
+** Between 0, and 0x1FFFFF.
+* Transaction Position
+** Between 0, and 0x1FFF.
+
+This application display format produces a 14-character data part starting with the character ‘r’.
+* Separators MUST be added as shown: tx1:rXXX-XXXX-XXXX-XX.
+{| class="wikitable"
+|'''Format ‘0’'''
+|'''Bit'''
+|'''Bit(s)'''
+|'''14 Character Encoding = 40bit + 30bit checksum'''
+|-
+|Magic
+|0 – 4
+|5
+|0x3
+|-
+|Version
+|5
+|1
+|Must be set to 0x0
+|-
+|Block Height
+|6 – 26
+|21
+|From the Genesis Block to about Year 2048
+|-
+|Tx Position
+|27 – 39
+|13
+|8191 transactions. (A full Bitcoin Block without Hard  Fork).
+|}
```

Comment:
I can understand the desire to keep all reference strings to the nice 14-character version by keeping the data payload to these 40 bits, but it seems to place artificial limitations on the format (year 2048 & 8191 transactions). I also understand that this might be addressed with Version 1 encoding. But current blocks are not that far from having 8191 transactions.

You could go with a variable-length encoding similar to Bitcoin's variable ints and gain the benefit of having a format that will work for very large blocks and the very far future.

Also, the Bech32 reference libraries allow encoding from byte arrays into the base-5 arrays native to Bech32. It seems like bit-packing to these 40 bits might be overkill. As an alternative you could have one bit-packed byte to start:
```
# First two bits are the protocol version, supporting values 0-3
V = ((protocol version) & 0x03) << 6
# Next two bits are magic for the blockchain
# 0x00 = Bitcoin
# 0x01 = Testnet3
# 0x02 = Byte1 is another coin's magic code (gives 256 options)
# 0x03 = Byte1-2 is treated as the coin magic code (gives 65280 more options)
M = (magic & 0x03) << 4
# Next two bits are the byte length of the block reference
B = ((byte length of block reference) & 0x03) << 2
# Final two bits are the byte length of the transaction index
T = ((byte length of transaction index) & 0x03)
# Assemble into the first byte
Byte0 = V | M | B | T
```
This gives you up to 3 bytes for each block and transaction reference, which is 16.7 M blocks, or year 2336, and 16.7 M transaction slots.
```
Data part: [Byte0][optional magic bytes 1-2][block reference bytes][tx reference bytes]
```
So the shortest data part would have 3 bytes in it, with the reference version `0` genesis coinbase transaction having data part `0x050000`.

I know this is a departure from your vision, but it would be much more flexible for the long term (in my opinion).